tests/Schema/TableTest.php

```diff
@@ -852,4 +852,18 @@ public function testRemoveUniqueConstraintUnknownNameThrowsException(): void
 
         $table->removeUniqueConstraint('unique_constraint');
     }
+
+    public function testPrimaryKeyWithCaptialization(): void
+    {
+        $table = new Table('foo');
+        $table->addColumn('fooID', 'integer');
+        $table->addColumn('bar_id', 'integer');
+        $table->setPrimaryKey(['bar_id', 'fooID']);
+
+        $primaryKey = $table->getPrimaryKeyColumns();
+
+        self::assertCount(2, $primaryKey);
+        self::assertArrayHasKey('fooid', $primaryKey);
```

Comment:
Why is it expected that the key is lower-cased? Column names are sensitive on some platforms and with some settings, so this might be the problem in the first place.

Reply:
If you ask me, it's not expected :) 

The problem is that the columnames are already "normalized", meaning that they are stored in the lowercased form. The original name is not stored. 

I suppose it would be wise to do the normalization later, when needed, so the original columnnames are stored and getPrimaryKeyColumns can do simple lookup. That means that all other functions need to do the sanitazion on demand. Since this seems like a whole lot more work I do not want to do that unless I know that's the correct direction. Otherwise, it would be a waste of my time to do the change, and yours of checking it.

Reply:
Agree. Let's leave the "normalization" in place for now.